commands/user.go

```diff
@@ -719,3 +800,74 @@ func convertBotToUser(c client.Client, cmd *cobra.Command, userArgs []string) er
 
 	return nil
 }
+
+func migrateAuthCmdF(c client.Client, cmd *cobra.Command, userArgs []string) error {
+	if userArgs[1] == "saml" {
+		return migrateAuthToSamlCmdF(c, cmd, userArgs)
+	}
+	return migrateAuthToLdapCmdF(c, cmd, userArgs)
+}
+
+func migrateAuthToSamlCmdF(c client.Client, cmd *cobra.Command, userArgs []string) error {
+	fromAuth := userArgs[0]
+	auto, _ := cmd.Flags().GetBool("auto")
+	confirm, _ := cmd.Flags().GetBool("confirm")
+	if auto && !confirm {
+		var confirm string
+		printer.Print("You are about to perform an automatic \"" + fromAuth + " to saml\" migration. This must only be done if your current Mattermost users with " + fromAuth + " auth have the same username and email in your SAML service. Otherwise, provide the usernames and emails from your SAML Service using the \"users file\" without the \"--auto\" option.\n\nDo you want to proceed with automatic migration anyway? (YES/NO):")
+		fmt.Scanln(&confirm)
+
+		if confirm != "YES" {
+			return errors.New("aborted: You did not answer YES exactly, in all capitals")
+		}
+	}
+
+	matches := map[string]string{}
+	if !auto {
+		matchesFile := userArgs[2]
+
+		file, err := ioutil.ReadFile(matchesFile)
+		if err != nil {
+			return errors.New("invalid users file")
+		}
+		if err := json.Unmarshal(file, &matches); err != nil {
+			return errors.New("invalid users file")
```

Comment:
Minor nit:  would be nice to differentiate between "can't read file" and "invalid JSON"

Reply:
Good point 👍